maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/RelocatedArtifact.java

```diff
@@ -86,6 +86,39 @@ public String getVersion()
         }
     }
 
+    @Override
+    public Artifact setVersion( String version )
+    {
+         String current = getVersion();
+         if ( current.equals( version ) || ( version == null && current.length() <= 0 ) )
```

Comment:
How could the length of `current` be `< 0`?

Reply:
It can't. You see this pattern everywhere in Maven space. I simply copied the code from the parent class. I intentionally did not want to change it in this spot. It should be changed all at once everywhere.

Reply:
It's the first time I encounter it... Just being curious

Reply:
Negative (black) magic.